alacritty_terminal/src/config/mod.rs

```diff
@@ -242,20 +242,58 @@ impl Default for EscapeChars {
 }
 
 #[serde(default)]
-#[derive(Deserialize, Copy, Clone, Debug, Default, PartialEq, Eq)]
+#[derive(Deserialize, Copy, Clone, Debug, Default, PartialEq)]
 pub struct Cursor {
     #[serde(deserialize_with = "failure_default")]
     pub style: CursorStyle,
     #[serde(deserialize_with = "option_explicit_none")]
     pub vi_mode_style: Option<CursorStyle>,
     #[serde(deserialize_with = "failure_default")]
+    thickness: CursorThickness,
+    #[serde(deserialize_with = "failure_default")]
     unfocused_hollow: DefaultTrueBool,
 }
 
 impl Cursor {
+    #[inline]
     pub fn unfocused_hollow(self) -> bool {
         self.unfocused_hollow.0
     }
+
+    #[inline]
+    pub fn thickness(self) -> f32 {
+        self.thickness.0
+    }
+}
+/// Wrapper around f32 that represents a cursor thickness between 0.0 and 1.0
+#[derive(Clone, Copy, Debug, PartialEq)]
+struct CursorThickness(f32);
```

Comment:
I'm in favor of using `f32`, since it's OpenGL stuff for alpha is using f32 all over the place, and I'm in favor of casting f32 to f64 than downcasting.